src/js/port_handler.js

```diff
@@ -174,9 +174,15 @@ PortHandler.detectPort = function(currentPorts) {
         if (GUI.auto_connect && !GUI.connecting_to && !GUI.connected_to) {
             // start connect procedure. We need firmware flasher protection over here
             if (GUI.active_tab !== 'firmware_flasher') {
-                GUI.timeout_add('auto-connect_timeout', function () {
-                    $('div#header_btns a.connect').click();
-                }, 100); // timeout so bus have time to initialize after being detected by the system
+                let connectionTimeout = 100;
+                ConfigStorage.get('connectionTimeout', function (result) {
```

Comment:
Looks like we are using the same setting here and in `serial_backend.js` to set two different timeouts, one of which is 15 times as long as the other by default - why are we setting it to the same, this does not seem to be making sense.